ignite/metrics/metric.py

```diff
@@ -548,9 +548,15 @@ def another_wrapper(self: Metric, *args: Any, **kwargs: Any) -> Callable:
             if len(attrs) > 0 and not self._is_reduced:
                 if ws > 1:
                     for attr in attrs:
+                        if ":" in attr:
+                            idx = attr.index(":")
+                            op = attr[idx + 1 :].strip().upper()
+                            attr = attr[:idx]
+                        else:
+                            op = "SUM"
```

Comment:
Same here, maybe if no op is provided, we should call `idist.all_reduce(t)` and use the default of `all_reduce`.

Reply:
I will work on tests just after we agree on the implementation
> * test all the combination of op

That's already done in `torch.distirubted` but i can do that in `sync_all_reduce()`, what do you think?
> * catch the error if a wrong op (for instance `"var:fakeop"`, `"var:"`, etc.)

Reply:
Can you please explain your point more?

Reply:
If no op is provided, we could have a default as you did. Another option is to use the `idist.all_reduce` without `op` argument.
```python
for attr in attrs:
    op_kwargs = {}
    if ":" in attr:
        idx = attr.index(":")
        op = attr[idx + 1 :]
        valid_ops = ["min", "max", "sum", "prod"]
        if op not in valid_ops:
            raise ValueError(f"Reduction operation is not valid (expected : {valid_ops}, got: {op}")
        attr = attr[:idx]
        op_kwargs["op"] = op
    t = getattr(self, attr, None)
    if t is not None:
        t = idist.all_reduce(t, **op_kwargs)
        self._is_reduced = True
        setattr(self, attr, t)
```
Thoughts on that ?

Reply:
Seems cleaner, thanks for the thoughts